src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/LoggerMessageGenerator.Parser.cs

```diff
@@ -651,10 +654,50 @@ private static string GenerateClassName(TypeDeclarationSyntax typeDeclaration)
                                 return (null, true);
                             }
                         }
+
+                        shadowedNames.Add(fs.Name);
```

Comment:
>shadowedNames.Add(fs.Name); [](http://example.com/codeflow?start=24&length=27)

should have this with `else`? we should carry the shadowed names that not are loggerSymbol.

Reply:
It doesn't make any difference for the behavior, since when a logger field is found, the primary constructor code isn't run, but you're right that we may as well avoid the allocation.